ca/server.go

```diff
@@ -565,6 +576,37 @@ func (s *Server) isRunning() bool {
 	return true
 }
 
+// saveAndFilterExternalCAURLS saves a copy of the external CAs to the server, and also returns a list of external CA
+// urls filtered by the desired cert. This function assumes that something else has called s.signingMu.Lock() and
+// will also unlock it
+func (s *Server) saveAndFilterExternalCAURLS(ctx context.Context, desiredCert, defaultCert []byte, apiExternalCAs []*api.ExternalCA) (urls []string) {
+	desiredCert = NormalizePEMs(desiredCert)
+	s.lastSeenExternalCAs = make([]*api.ExternalCA, len(apiExternalCAs))
```

Comment:
I have a slight preference for going back to the old code's `s.lastSeenExternalCAs = cluster.Spec.CAConfig.Copy().ExternalCAs` expression, moved outside this function.

The reason is that saving the URL list and filtering it are different operations, so I think keeping them separate is more understandable.